### PR TITLE
Feature/issue #101 update log4j library

### PR DESCRIPTION
**Pull request by valliman**
#101 Upgrade to Log4j2.

I deleted most of the manual logger configuration in tests. I would suggest to provide a log4j2-test.xml or similar for such purposes. Is there an easy way to verify that everything works via the CI server before performing the changes? (Sadly my local setup is currently somehow broken and my machine refuses to open any ports... )

### CODE REVIEW

**Comment by deruelle** [comment]
Thanks @valliman ! @jaimecasero can you review ?

@valliman can you sign the Contributor License Agreement at http://www.telestax.com/open-source/#Contribute so we can accept your contribution ?

**Comment by valliman** [comment]
I've signed the CLA today.

File: src/examples/cancel/Shootist.java [hunk 1]
```diff
@@ -56,18 +76,12 @@
...
-
-
-
     public void processRequest(RequestEvent requestReceivedEvent) {
         Request request = requestReceivedEvent.getRequest();
         ServerTransaction serverTransactionId = requestReceivedEvent
                 .getServerTransaction();
 
-        logger.info("\n\nRequest " + request.getMethod()
+        LOG.info("\n\nRequest " + request.getMethod()
```
Comment (valliman) [inline src/examples/cancel/Shootist.java]: A private static final field should be cased like that. I know this spreads through the test classes and I can't follow your argument concerning the log4j API.

File: log4j2.xml [hunk 2]
```diff
@@ -5,17 +5,20 @@
...
     <Appenders>
-        <Console name="A1">
-            <SimpleLayout/>
+        <Console name="Console">
+            <PatternLayout pattern="%d [%-6p] %c - %m%n"/>
         </Console>
-        <File name="A2" fileName="${filename}">
-            <SimpleLayout/>
+        <File name="File" fileName="${filename}">
+            <PatternLayout pattern="%d [%-6p] %c - %m%n"/>
```
Comment (valliman) [inline log4j2.xml]: I changed the pattern to match the previously used simpleLayout

File: src/test/unit/gov/nist/javax/sip/stack/WebsocketSelfTest.java [hunk 3]
```diff
@@ -316,6 +316,13 @@ public void terminate() {
         private ListeningPoint udpListeningPoint;
         private Dialog dialog;
         public boolean okByeReceived;
+        public String initialInvitePayload = "v=0\r\n"
+                + "o=4855 13760799956958020 13760799956958020"
+                + " IN IP4  129.6.55.78\r\n" + "s=mysession session\r\n"
+                + "p=+46 8 52018010\r\n" + "c=IN IP4  129.6.55.78\r\n"
+                + "t=0 0\r\n" + "m=audio 6022 RTP/AVP 0 4 18\r\n"
+                + "a=rtpmap:0 PCMU/8000\r\n" + "a=rtpmap:4 G723/8000\r\n"
+                + "a=rtpmap:18 G729A/8000\r\n" + "a=ptime:20\r\n";
```
Comment (valliman) [inline src/test/unit/gov/nist/javax/sip/stack/WebsocketSelfTest.java]: Oh dear, apparently I did an auto-formatting here. I reverted that.

File: src/examples/cancel/Shootist.java (hunk 1 above)
Comment (jaimecasero) [inline src/examples/cancel/Shootist.java]: I agree there are conventions to set var names, but JAINSIP stack is an old guy, there are lots of style problem everywhere. I would perefer to modify as less files as possible, to make the commit targetted to log4j upgrade. If you are concerned about code convention, we can open a new issue, and cover code convention problems regarding log4j and more ( there are plenty of them).

My point on Log4jAPI was that I would be ok to modify 170 files for the log4j2 upgrade, if they really introduce an enhacement in terms of logging usage, but just changing the Logger var name is not reason enough to change those 170 files IMO.

Comment (valliman) [inline src/examples/cancel/Shootist.java]: I will try my best to reduce it as much as possible, as I removed in plenty of files the manual logger config I am quite unsure if the amount of files with changes will go down a lot, however let's hope for the best. I will ping you again once I've made those changes.

Concerning code-style: I would propose to maven-ize the project first and then introduce a common code style (which is enforced by the maven-checkstyle-plugin).

BR Stefan

Comment (jaimecasero) [inline src/examples/cancel/Shootist.java]: i bet you refer to test classes that were configuring log4j, that is fine but those should be less. The logger var rename basically changed all the files (at least all classes using logging) which was too much

Regarding code-style and mavenization I totally agree. Mavenization is first item on our techDebt backlog. We are doing some efforts on this, like allowing testSuite to be run in parallel, and we will release this soon. Things like using checkstyle or enforcer plugins are considered alreadt, im happy you are aligned there. Again we may cover this in separate issue, are you interested in contributing on that side (mavenization)?

**Comment by valliman** [comment]
Hi, 
I re-worked the log4j2 upgrade, to use the log4j 1.2 API and I've reverted the silly logger renaming change.

**Comment by jaimecasero** [comment]
@valliman  can you just start form a fresh PR with minimum commits/changes as discussed? I still can see lots of files changed  (146) some of them just becuase of indentation fix. We cant merge this as it is now...

**Pull request closed without merging**